src/transformers/pipelines/base.py

```diff
@@ -1211,6 +1255,13 @@ def iterate(self, inputs, preprocess_params, forward_params, postprocess_params)
             yield self.run_single(input_, preprocess_params, forward_params, postprocess_params)
 
 
+Pipeline.push_to_hub = copy_func(Pipeline.push_to_hub)
+if Pipeline.push_to_hub.__doc__ is not None:
+    Pipeline.push_to_hub.__doc__ = Pipeline.push_to_hub.__doc__.format(
+        object="pipe", object_class="pipeline", object_files="pipeline file"
+    ).replace(".from_pretrained", "")
```

Comment:
This bit confuses me - since you're inheriting `PushToHubMixin.push_to_hub`, `__doc__` should always be defined, right? I can see it's a copy of the same code for the other classes that inherit from `PushToHubMixin`, though, I'm just not sure why it's coded this way. Not a blocker, just a comment!

Reply:
since other people used this method to copy the docs i chose to use the same one as them, to stay in the same page as them, just to avoid straying too much from the norm

Reply:
Yep, that's totally fine! I was just pointing out my own confusion, I guess

Reply:
true that, i think the reason for them using this annotation is that they only need to change one method (the original one) to change the docs for all of the other classes using it.
meaning one changes all, which is a really nice approach